src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,75 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier implements SideCalculator {
+    private static final int INDEX_BOUND = 5;
+    private static final int DEFAULT_LENGTH = 10;
+    private static final String DEFAULT_COLOR = Color.WHITE.name();
+    private static final Random index = new Random();
+
+    // Initializing fields of object using constructor looks better,
+    // but also takes more code at this example
+    private Circle getPrepareCircle() {
+        Circle circle = new Circle();
+        circle.setRadius(side());
```

Comment:
```suggestion
    private Circle getPrepareCircle() {
        Circle circle = new Circle(color, side);
```
not fixed, you still use setters, all of the figures can be initialized in one line, if you create necessary constructors in each class. You can have a constructor for color in Figure superclass, and call super(color) in subclasses